Test Cert_9_2_11_MasterKey is Unreliable

I am getting failures in Cert_9_2_11_MasterKey.py fairly consistently on my master branch, which I keep as a clean clone of openthread/master. See failure [here](https://travis-ci.org/nibanks/openthread/jobs/174272023).

```
======================================================================
FAIL: test (__main__.Cert_9_2_11_MasterKey)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/nibanks/openthread/examples/../tests/scripts/thread-cert/Cert_9_2_11_MasterKey.py", line 150, in test
    self.assertEqual(self.nodes[SED1].get_masterkey(), KEY1)
AssertionError: u'0f0e0d0c0b0a09080706050403020100' != '000102030405060708090a0b0c0d0e0f'

----------------------------------------------------------------------
Ran 1 test in 58.407s
```

--- a/src/core/thread/mesh_forwarder.cpp
+++ b/src/core/thread/mesh_forwarder.cpp
@@ -1780,8 +1780,6 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const Thre
     child->mLastHeard = Timer::GetNow();
     child->mLinkFailures = 0;
 
-    mMle.HandleMacDataRequest(*child);
-
     if (child->mQueuedIndirectMessageCnt > 0)
     {
         child->mDataRequest = true;

--- a/src/core/thread/mle_router.cpp
+++ b/src/core/thread/mle_router.cpp
@@ -2276,6 +2276,35 @@ ThreadError MleRouter::HandleNetworkDataUpdateRouter(void)
 
     SendDataResponse(destination, tlvs, sizeof(tlvs));
 
+    for (uint8_t i = 0; i < mMaxChildrenAllowed; i++)
+    {
+        Child *child = &mChildren[i];
+
+        if (child->mState != Neighbor::kStateValid || (child->mMode & ModeTlv::kModeRxOnWhenIdle) != 0)
+        {
+            continue;
+        }
+
+        memset(&destination, 0, sizeof(destination));
+        destination.mFields.m16[0] = HostSwap16(0xfe80);
+        destination.SetIid(child->mMacAddr);
+
+        if (child->mMode & ModeTlv::kModeFullNetworkData)
+        {
+            if (child->mNetworkDataVersion != mNetworkData.GetVersion())
+            {
+                SendDataResponse(destination, tlvs, sizeof(tlvs));
+            }
+        }
+        else
+        {
+            if (child->mNetworkDataVersion != mNetworkData.GetStableVersion())
+            {
+                SendDataResponse(destination, tlvs, sizeof(tlvs));
+            }
+        }
+    }
+
 exit:
     return kThreadError_None;
 }
@@ -2857,36 +2886,6 @@ ThreadError MleRouter::SetPreferredRouterId(uint8_t aRouterId)
 
 exit:
     return error;
-}
-
-void MleRouter::HandleMacDataRequest(const Child &aChild)
-{
-    static const uint8_t tlvs[] = {Tlv::kLeaderData, Tlv::kNetworkData};
-    Ip6::Address destination;
-
-    VerifyOrExit(aChild.mState == Neighbor::kStateValid && (aChild.mMode & ModeTlv::kModeRxOnWhenIdle) == 0, ;);
-
-    memset(&destination, 0, sizeof(destination));
-    destination.mFields.m16[0] = HostSwap16(0xfe80);
-    destination.SetIid(aChild.mMacAddr);
-
-    if (aChild.mMode & ModeTlv::kModeFullNetworkData)
-    {
-        if (aChild.mNetworkDataVersion != mNetworkData.GetVersion())
-        {
-            SendDataResponse(destination, tlvs, sizeof(tlvs));
-        }
-    }
-    else
-    {
-        if (aChild.mNetworkDataVersion != mNetworkData.GetStableVersion())
-        {
-            SendDataResponse(destination, tlvs, sizeof(tlvs));
-        }
-    }
-
-exit:
-    {}
 }
 
 Router *MleRouter::GetRouters(uint8_t *aNumRouters)

--- a/src/core/thread/mle_router.hpp
+++ b/src/core/thread/mle_router.hpp
@@ -560,14 +560,6 @@ public:
     ThreadError GetRouterInfo(uint16_t aRouterId, otRouterInfo &aRouterInfo);
 
     /**
-     * This method handles MAC Data Poll messages.
-     *
-     * @param[in]  aChild  The Child that sent the MAC Data Poll message.
-     *
-     */
-    void HandleMacDataRequest(const Child &aChild);
-
-    /**
      * This method indicates whether or not the given Thread partition attributes are preferred.
      *
      * @param[in]  aSingletonA   Whether or not the Thread Partition A has a single router.

